cmd/operator-sdk/main.go

```diff
@@ -81,3 +84,66 @@ func main() {
 		os.Exit(1)
 	}
 }
+
+func checkDepManagerForCmd(cmd *cobra.Command) (err error) {
+	// Do not perform this check if wd is not in the project root,
+	// as some sub-commands might not require project root.
+	if err := projutil.CheckProjectRoot(); err != nil {
+		return nil
+	}
+
+	var dm projutil.DepManagerType
+	switch cmd.Name() {
+	case "new", "migrate":
+		// Do not perform this check if the new project is non-Go, as they do not
+		// have (Go) dep managers.
+		if cmd.Name() == "new" {
+			projType, err := cmd.Flags().GetString("type")
+			if err != nil {
+				return err
+			}
+			if projType != "go" {
+				return nil
+			}
+		}
+		// "new" and "migrate" commands are for projects that establish which
+		// dep manager to use. "new" should not be called if we're in the project
+		// root but could be, so check it anyway.
+		dmStr, err := cmd.Flags().GetString("dep-manager")
+		if err != nil {
+			return err
+		}
+		dm = projutil.DepManagerType(dmStr)
+	default:
```

Comment:
Should we also always skip the dep-manger check for the `completion`, `help`, and `version` subcommands?

Reply:
Yes we should.